lib/puppet/provider/tomcat_install/linux.rb

```diff
@@ -0,0 +1,98 @@
+require 'fileutils'
+require 'rubygems/package'
+require 'zlib'
+require 'open-uri'
+
+Puppet::Type.type(:tomcat_install).provide(:linux) do
+  def initialize(value={})
+    super(value)
+  end
+
+  def get_tomcat_version(catalina_home)
+    return nil unless File.exist?("#{catalina_home}/RELEASE-NOTES")
+
+    version = nil
+
+    File.open("#{catalina_home}/RELEASE-NOTES", 'r').each_line do |li|
+      if li[/Version \d{1,2}\.\d{1,2}\.\d{1,2}/]
+        version = li[/\d{1,2}\.\d{1,2}\.\d{1,2}/]
+        break
+      end
+    end
+
+    version
+  end
+
+  def exists?
+    get_tomcat_version(resource[:catalina_home]) == resource[:version]
+  end
+
+  def destroy
+    FileUtils.rm_r resource[:catalina_home] if Dir.exist?(resource[:catalina_home])
+  end
+
+  def create
+    destroy
+
+    tomcat_major = resource[:version].split('.')[0]
+
+    tomcat_urls = [
+      "https://www.apache.org/dyn/closer.cgi?action=download&filename=tomcat/tomcat-#{tomcat_major}/v#{resource[:version]}/bin/apache-tomcat-#{resource[:version]}.tar.gz",
+      "https://www-us.apache.org/dist/tomcat/tomcat-#{tomcat_major}/v#{resource[:version]}/bin/apache-tomcat-#{resource[:version]}.tar.gz",
+      "https://www.apache.org/dist/tomcat/tomcat-#{tomcat_major}/v#{resource[:version]}/bin/apache-tomcat-#{resource[:version]}.tar.gz",
+      "https://archive.apache.org/dist/tomcat/tomcat-#{tomcat_major}/v#{resource[:version]}/bin/apache-tomcat-#{resource[:version]}.tar.gz"
+    ].freeze
+
+    FileUtils.mkdir_p resource[:catalina_home]
+
+    tomcat_url_index = -1
+
+    # Download the tomcat zip
+    tomcat_urls.each do |url|
+      tomcat_url_index += 1
```

Comment:
Ruby has an `each_with_index` function that will pass in the index. Though it doesn't appear to be needed at all?

Reply:
So you can delete this line.